README_WAKE.adoc

```diff
@@ -0,0 +1,154 @@
+
+=== Return Type
+
+The return type of of `runRocketChipGenerator` is a `RocketChipGeneratorOutputs`.
+It provides the following accessors for its contents:
+
+----
+global def getRocketChipGeneratorOutputsDTS             = ...
+global def getRocketChipGeneratorOutputsFirrtlFile      = ...
+global def getRocketChipGeneratorOutputsFirrtlAnnoFile  = ...
+global def getRocketChipGeneratorOutputsRomConf         = ...
+global def getRocketChipGeneratorOutputsAllOutputs      = ...
+global def getRocketChipGeneratorOutputsInputOptions    = ...
+global def getRocketChipGeneratorOutputsObjectModelFile = ...
+----
+
+== Run Simulations
+
+The Wake rules for simulations are in the https://github.com/sifive/api-generator-sifive[API Generator SiFive] repository.
+
+You can add it to an existing Rocket Chip workspace:
+
+----
+$ wit add-pkg git@github.com:sifive/api-generator-sifive.git
+$ wit update
+----
+
+This will pull in `api-generator-sifive` as well as additional dependencies.
+
+== Tips and Tricks
+
+See the https://github.com/sifive/wake/tree/a0d99f15cf806f917d64714934d337b392d36f26/share/doc/wake[Wake `share/doc` directory] for documentation.
+Note that this link is to a newer commit than 0.15.1 because more documentation has been written since the release.
+Wake is still evolving so its hard for everything to be 100% synced, but the linked documentation should be compatible at time of writing.
+
+=== Specific recommended reading:
+
+* https://github.com/sifive/wit/blob/v0.10.1/share/doc/wit/tutorial.md[Wit tutorial]
+* https://github.com/sifive/wake/blob/v0.15.1/share/doc/wake/tutorial.md[Wake tutorial]
+* https://github.com/sifive/wake/blob/a0d99f15cf806f917d64714934d337b392d36f26/share/doc/wake/quickref.md[Wake Quickref]
+* https://github.com/sifive/wake/blob/a0d99f15cf806f917d64714934d337b392d36f26/share/doc/wake/tour/tuples.adoc[Wake Tuple Docs]
+* https://github.com/sifive/api-scala-sifive/tree/aafc0a430db848f063ed906ba84b1ddfab56ca29[API Scala SiFive documentation]
+* https://sifive.github.io/wake/[Annotated Wake Standard Library]
+
+Another useful thing to look at is https://github.com/sifive/block-pio-sifive[Block PIO SiFive] which is an example repository for onboarding Verilog IP to Rocket Chip-based designs
+
+=== Wake tips
+
+* Wake has https://github.com/sifive/wake/tree/a0d99f15cf806f917d64714934d337b392d36f26/share/doc/wake/syntax[syntax highlighting] for Vim, Emacs, and Joe
+
+* Be sure to check out the Wake CLI `wake --help`.
+
+* Turn on verbose with `-v`. You can use this to see the type of a function, eg. `wake -v map`.
+
+* See all global variables with `-g`, this can be combined with `grep` to see available functions for a given type, try
+----
+$ wake -g | grep ScalaModule
+----
+
+* You can see the job that created a file with `wake -o <file>`, and jobs that use a file as input with `wake -i <file>`
+
+* You can combine `-o` and `-i` with `-s` to create an exectuable bash script that runs a job.
```

Comment:
Might be worth mentioning that the executable shell script includes things like setting up all the environment variables.

Reply:
Agreed